### PR TITLE
DAOS-14491: Retain support for phase-1 DAV heap

### PR DESCRIPTION
**Pull request by sherintg**
A copy of the phase-1 DAV allocator with some cleanup is moved to the subdirectory src/common/dav_v1. This allocator is built as a standalone shared library and linked to the libdaos_common_pmem library.  The umem will now support one more mode DAOS_MD_BMEM_V1 which results in umem instance using phase-1 DAV allocator interface and layout.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: utils/rpms/daos.spec
Comment (brianjmurrell) [inline utils/rpms/daos.spec:1]: I understand this is on a feature branch, but just keep in mind that any update to this file that results in a packaging difference needs to include a `Version:` or `Release:` bump and a `%changelog` entry.

You probably also need to update `debian/` to account for this new file and update `debian/changelog` accordingly.

Comment (sherintg) [reply utils/rpms/daos.spec:1]: thanks @brianjmurrell. I will update them too. Is there a documentation that I can refer to?

Comment (brianjmurrell) [reply utils/rpms/daos.spec:1]: Probably just looking at the history of the individual files (https://github.com/daos-stack/daos/commits/master/utils/rpms/daos.spec, https://github.com/daos-stack/daos/commits/master/debian/changelog, https://github.com/daos-stack/daos/commits/master/debian/control) will tell you all you need to know.  You should be able to find quite easily a commit that updates all three due to adding a library (or any other file).

**Review by jolivier23 — APPROVED** [review]

File: src/common/dav/bucket.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2022, Intel Corporation */
```
Comment (brianjmurrell) [inline src/common/dav/bucket.c:2]:
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```
yes?

File: src/common/dav/bucket.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2021, Intel Corporation */
```
Comment (brianjmurrell) [inline src/common/dav/bucket.h:2]:
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```

File: src/common/dav/dav.h [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2022, Intel Corporation */
```
Comment (brianjmurrell) [inline src/common/dav/dav.h:2]:
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```

File: src/common/dav/heap.c [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2022, Intel Corporation */
```
Comment (brianjmurrell) [inline src/common/dav/heap.c:2]:
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```

I won't comment on any more erroneous copyright date changes but please review all of your changes for this and correct.

File: utils/rpms/daos.spec [hunk 5]
```diff
@@ -585,6 +586,10 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Wed Oct 16 2023 Sherin T George <sherin-t.george@hpe.com> 2.5.100-10
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:589]:
```suggestion
* Mon Oct 16 2023 Sherin T George <sherin-t.george@hpe.com> 2.5.100-10
```
perhaps?

**Review by brianjmurrell — DISMISSED** [review]

**Comment by sherintg** [comment]
@brianjmurrell I have addressed your review comments. Apologies for making multiple force pulls after your initial review comments. There was a major design change requested and I had to redo the entire coding from the beginning.

**Review by brianjmurrell — COMMENTED** [review]
Packaging changes look good.  I will leave +1 slots for those more familiar with the bulk of the changes in this PR.

File: Jenkinsfile [hunk 6]
```diff
@@ -956,7 +956,7 @@ pipeline {
                     }
                     steps {
                         job_step_update(
-                            unitTest(timeout_time: 60,
+                            unitTest(timeout_time: 120,
```
Comment (NiuYawei) [inline Jenkinsfile:959]: What's the change for?

**Review by NiuYawei — APPROVED** [review]

**Pull request merged**